src/main/java/org/opentripplanner/graph_builder/module/osm/WayPropertySet.java

```diff
@@ -70,11 +81,112 @@ public WayPropertySet() {
         maxSpeedPattern = Pattern.compile("^([0-9][\\.0-9]+?)(?:[ ]?(kmh|km/h|kmph|kph|mph|knots))?$");
     }
 
+    /**
+     * Initializes lookups and Indexes for quick lookup of applicable values for way properties and car speeds. This
+     * function must be called before looking up way properties and car speeds for ways in order for those caches to be
+     * properly initialized.
+     */
+    public void index() {
+        // index for way properties
+        wayPropertyLookup = new HashMap<>();
+        possibleWayPropertyTagValues = new HashSet<>();
+        for (WayPropertyPicker wayProperty : wayProperties) {
+            for (P2<String> kvpair : wayProperty.getSpecifier().kvpairs) {
+                possibleWayPropertyTagValues.add(String.format("%s=%s", kvpair.first, kvpair.second));
+            }
+        }
+
+        // index for car speed lookup
+        carSpeedLookup = new HashMap<>();
+        possibleCarSpeedTagValues = new HashSet<>();
+        for (SpeedPicker speedPicker : speedPickers) {
+            for (P2<String> kvpair : speedPicker.specifier.kvpairs) {
+                possibleCarSpeedTagValues.add(String.format("%s=%s", kvpair.first, kvpair.second));
+            }
+        }
```

Comment:
These two blocks are essentially the same and could be refactored.

Reply:
I know that it looks similar, but it's just slightly different and there are only 2 implementations of this.